backend/src/main/kotlin/org/loculus/backend/config/SecurityConfig.kt

```diff
@@ -58,6 +58,7 @@ class SecurityConfig {
         "/get-seqset-cited-by-publication",
         "/get-author",
         "/*/get-released-data",
+        "/files/get/**",
```

Comment:
Should `/files/get/**` be public also for not-yet-released files?

Reply:
Yes? In that we handle the authentication in the endpoint itself

Reply:
Yes, in the endpoint we check if the `User` is an `AuthenticatedUser`, and that they are allowed to access the file. 

        val isPublic = filesDatabaseService.isFilePublic(fileId)!!
        if (!isPublic) {
            if (user is AuthenticatedUser) {
                accessionPreconditionValidator.validate {
                    thatAccessionVersionExists(accessionVersion)
                        .andThatUserIsAllowedToEditSequenceEntries(user)
                }
            } else {
                throw ForbiddenException("Authentication needed: the file is not public")
            }
        }
        
The endpoint is public in that you _can_ access it without authentication and in some cases you also will get a file back.

Reply:
Because otherwise it would be impossible for any files to be accessed without auth

Reply:
Thanks! 

Adding more context for myself and anyone else is curious: `"/files/get/**"` is a get endpoint whereas `submit` is a post endpoint. In the security config (securityFilterChain) all POST requests fall into the `auth.anyRequest().authenticated()` bucket and are only let in with authentication whereas GET requests are handled with `auth.requestMatchers(HttpMethod.GET, *getEndpointsThatArePublic).permitAll()` - if  `"/files/get/**"` is not added here it will also fall through to `auth.anyRequest().authenticated()` bucket and require auth to be accessed